### PR TITLE
Re-enable community membership selection

### LINKED ISSUE #12548: Re-enable open-membership communities
As a user
I want to be able to create an open-membership community
So that I can embed my community on my website

### Description

We should re-enable open-membership communities (not the default option), so that we can create them and they can be embedded in dapps.

cc @D4nte

### PR DESCRIPTION
**Pull request by Jonathansoufer**
Re-enables community creation based on 2 types of membership:

- Require approval
- No requirement (Open Membership Community)

fixes #12548

### Summary

This PR re-enables a feature which allows users to define three levels of community membership upon community creation. This feature was disabled by the issue #12147, and re-enabled now.
...

#### Platforms

- Android
- iOS

#### Areas that maybe impacted
community creation/edit

##### Functional

- community creation

### Steps to test
- Open the app
- Create a community & specify one of the levels of membership

Status: Ready

Screenshot 2021-12-16 at 15 24 05

### CODE REVIEW

**Comment by du82** [comment]
I think the options should be:
* Private = No new invites can be created, no one new can join, existing members can continue to talk or leave.
* Invite Only = Does not show up in public lists. Existing members can invite new people, who must go through admin approval prior to entry.
* Open Chat = Shows up in public lists. Anyone with a link can join without admin approval and talk instantly.

There should be a way to change between the three permission types after the community is created.

I think we also need per-chat permissions, so for example there can be an announcements channel that members can't talk in, but admins can. There also needs to be a way to share communities in private and group chats, likely through the way of chat extensions.

File: src/status_im/communities/core.cljs [hunk 1]
```diff
@@ -221,18 +221,21 @@
...
+    (let [params (cond-> {:name name
+                          :description description
+                          :membership (or membership constants/community-no-membership-access)
+                          :color (rand-nth colors/chat-colors)
+                          :image (string/replace-first (str image) #"file://" "")
+                          :imageAx 0
+                          :imageAy 0
+                          :imageBx crop-size
+                          :imageBy crop-size}
+                   (= membership constants/community-rule-ens-only)
```
Comment (cammellos) [inline src/status_im/communities/core.cljs]: I don't think we need to re-introduce this change at this time, we are not allowing for now ENS only communities

File: src/status_im/ui/screens/communities/profile.cljs [hunk 2]
```diff
@@ -51,7 +51,8 @@
                description]
               [quo/separator {:style {:margin-vertical 8}}]])
            [:<>
-            (let [link (communities/universal-link community-id)]
+            (let [link #_{:clj-kondo/ignore [:unresolved-var]}
```
Comment (cammellos) [inline src/status_im/ui/screens/communities/profile.cljs]: Just out of curiosity, did you have to disable the linter on this? it looks like it's being used, so the linter should not complain

Comment (Jonathansoufer) [reply src/status_im/ui/screens/communities/profile.cljs]: Thanks. This was used before by me and was left by mistake.

File: src/status_im/communities/core.cljs (hunk 1 above)
Comment (Jonathansoufer) [reply src/status_im/communities/core.cljs]: Roget that. Removed useless code from non necessary option on membership list.

File: src/status_im/communities/core.cljs [hunk 3]
```diff
@@ -221,18 +220,11 @@
...
-    (let [params {:name name
-                  :description description
-                  :membership constants/community-on-request-access
-                  :color (rand-nth colors/chat-colors)
-                  :image (string/replace-first (str image) #"file://" "")
-                  :imageAx 0
-                  :imageAy 0
-                  :imageBx crop-size
-                  :imageBy crop-size}]
+    (let [params ((assoc :membership constants/community-on-request-access
```
Comment (cammellos) [inline src/status_im/communities/core.cljs]: Mmh, not sure, I think this should be the same as before, no changes are required, `ens-only` communities are not used at the moment, so you can keep the same as develop

Comment (Jonathansoufer) [reply src/status_im/communities/core.cljs]: Roger that. Rolled back.

**Review by cammellos — APPROVED** [review]

**Comment by cammellos** [comment]
> I think the options should be:
> 
> * Private = No new invites can be created, no one new can join, existing members can continue to talk or leave.
> * Invite Only = Does not show up in public lists. Existing members can invite new people, who must go through admin approval prior to entry.
> * Open Chat = Shows up in public lists. Anyone with a link can join without admin approval and talk instantly.
> 
> There should be a way to change between the three permission types after the community is created.
> 
> I think we also need per-chat permissions, so for example there can be an announcements channel that members can't talk in, but admins can. There also needs to be a way to share communities in private and group chats, likely through the way of chat extensions.

Thanks @ducheng0 for the suggestion!

This is precisely what the protocol currently supports, though we want to move a bit slower in enabling all the options ( we did that before, but the UI/UX was a bit too confusing).

Hopefully slowly we will be adding a few options, without any UX detriment

**Comment by du82** [comment]
> > I think the options should be:
> > 
> > * Private = No new invites can be created, no one new can join, existing members can continue to talk or leave.
> > * Invite Only = Does not show up in public lists. Existing members can invite new people, who must go through admin approval prior to entry.
> > * Open Chat = Shows up in public lists. Anyone with a link can join without admin approval and talk instantly.
> > 
> > There should be a way to change between the three permission types after the community is created.
> > I think we also need per-chat permissions, so for example there can be an announcements channel that members can't talk in, but admins can. There also needs to be a way to share communities in private and group chats, likely through the way of chat extensions.
> 
> Thanks @ducheng0 for the suggestion!
> 
> This is precisely what the protocol currently supports, though we want to move a bit slower in enabling all the options ( we did that before, but the UI/UX was a bit too confusing).
> 
> Hopefully slowly we will be adding a few options, without any UX detriment

Another idea is to implement roles, so lets say only people with a "Content Creator" role or higher can invite people.
For communities ro succeed, I think there needs to be user-defined roles, like on Discord.

**Comment by qoqobolo** [comment]
Hey @Jonathansoufer, can you rebase it against the latest develop, please?

Also, could you clarify one moment, please? As I can see in the [feature request](https://github.com/status-im/status-react/issues/12548), only one community type needs to be re-enabled - `Open-membership communities`. This point was discussed on the last Mobile planning call as well. 

But the `Require invite from another member` was also added in this PR.
Just to make sure, is this type supposed to be re-enabled now?

File: src/status_im/ui/screens/communities/membership.cljs [hunk 4]
```diff
@@ -10,13 +10,13 @@
 (def options {constants/community-on-request-access
               {:title       :t/membership-approval
                :description :t/membership-approval-description}
-              constants/community-invitation-only-access
-              {:title       :t/membership-invite
-               :description :t/membership-invite-description}
+              ;; constants/community-invitation-only-access
```
Comment (shivekkhurana) [inline src/status_im/ui/screens/communities/membership.cljs]: Is this supposed to be commented?

**Review by shivekkhurana — APPROVED** [review]

**Comment by Jonathansoufer** [comment]
> Hey @Jonathansoufer, can you rebase it against the latest develop, please?
> 
> Also, could you clarify one moment, please? As I can see in the [feature request](https://github.com/status-im/status-react/issues/12548), only one community type needs to be re-enabled - `Open-membership communities`. This point was discussed on the last Mobile planning call as well.
> 
> But the `Require invite from another member` was also added in this PR. Just to make sure, is this type supposed to be re-enabled now?

As per a call session I did with Andrea, those two are the options which should be available to the users in this phase, since the open membership was already in place before, but hidden from the user.

Screenshot 2021-12-16 at 12 14 03

File: src/status_im/ui/screens/communities/membership.cljs (hunk 4 above)
Comment (Jonathansoufer) [reply src/status_im/ui/screens/communities/membership.cljs]: Yes. This can be a feature we bring back in future, but as for now will be hidden from users.

**Comment by qoqobolo** [comment]
@Jonathansoufer ahh okay :)
The case is that the description of this PR refers to **three** types (`Require approval`, `Require invite from another member`, `No requirement`). 

>Re-enables community creation based on 3 types of membership:
>- Require approval
>- Require invite from another member
>- No requirement (Open Membership Community)

>Summary
This PR re-enables a feature which allows users to define three levels of community membership upon community creation. This feature was disabled by the issue #12147, and re-enabled now.

Also in the video in the description (screenshot below), there are also three types.

## Updated.
Screenshot 2021-12-16 at 15 24 05


But in fact, there are only two of them in the PR itself (`Require approval` and `No requirement`) as expected.
That's why I asked you. Just misunderstanding :)
Could you edit the description for clarity, please?

**Comment by qoqobolo** [comment]
@Jonathansoufer thanks for the update!

### ISSUE 1: Can't create a community
The error occurs when pressing the `Create` button. 

Steps:
1. Install the `8bdd7be` build
2. Tap `+` on the home screen > `Communities` > `Create a community`
3. Fill in the `name` and `description` fields 
4. Tap `Create`

**OS:** Android, iOS

[geth.log](https://github.com/status-im/status-react/files/7734366/geth.log)

https://user-images.githubusercontent.com/67952253/146530275-7d1a9334-15a8-46af-975e-8a968192a258.MP4

**Comment by Jonathansoufer** [comment]
> @Jonathansoufer thanks for the update!
> 
> ### ISSUE 1: Can't create a community
> The error occurs when pressing the `Create` button.
> 
> Steps:
> 
> 1. Install the `8bdd7be` build
> 2. Tap `+` on the home screen > `Communities` > `Create a community`
> 3. Fill in the `name` and `description` fields
> 4. Tap `Create`
> 
> **OS:** Android, iOS
> 
> [geth.log](https://github.com/status-im/status-react/files/7734366/geth.log)
> 
>  IMG_1008.MP4

Got fixed. This was a bug introduced when I was rebasing the branch. Sorry. 

https://user-images.githubusercontent.com/44679989/146557904-f3b44ee5-b3a4-4b29-b162-49968a98eeba.mov

**Comment by churik** [comment]
@Jonathansoufer 
### ISSUE 2: can't see type of community on community info view

**Steps:**
1) create `require approval` community
2) create `no requirement` community
3) go to community info to see type of each of communities


**Expected result:**
can distinguish `require approval`  and  `no requirement`
**Actual result:**
IMAGE 2021-12-24 12:00:48
**OS:** IOS, Android

### ISSUE 3: community invite link flow is the same for both tyoes of communities

**Steps:**
1) User A: create `require approval` community
2) User A:  create `no requirement` community
3) User A:  go to community info for both of communities, copy and send into public chat community links
4) User B: join same public chat, wait until community links will be resolved and tap on both

**Expected result:**
For `require approval` community: can `Request access`
For `no requirement` community: can `Join`
**Actual result:**

https://user-images.githubusercontent.com/4557972/147347545-89cd62f3-c3a9-436b-89d9-94fbc632ac22.mp4


**OS:** IOS, Android

Basically I believe the same type of community is created (`require approval`) regardless of chosen option, as there is no difference in any tested flow between two of them.

**Comment by Jonathansoufer** [comment]
Hey @churik Issues 1 and 2 were already fixed and code is committed. For the Issue 3 I need some feedback from @cammellos or @flexsurfer once I am re-enabling the UI to allow users access features implemented already. @cammellos / @flexsurfer, could you help me to identify if createCommunity is taking into account membership on backend side?
